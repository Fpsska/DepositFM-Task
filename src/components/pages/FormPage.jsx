import React from "react";
import Form from "../form/Form";

const FormPage = () => {
    return (
        <>
            <Form />
        </>
    )
}

export default FormPage;