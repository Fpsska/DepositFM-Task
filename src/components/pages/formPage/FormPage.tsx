import React from 'react';

import Form from '../../form/Form';

const FormPage: React.FC = () => {
    return (
        <div className="form-page">
            <Form />
        </div>
    );
};

export default FormPage;