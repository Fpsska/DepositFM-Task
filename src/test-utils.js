import { render as rtlRender } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';

import { BrowserRouter } from 'react-router-dom';

import navSlice from './app/slices/navSlice';
import formSlice from './app/slices/formSlice';
import paletteSlice from './app/slices/paletteSlice';

// /. imports

const render = (
    ui,
    { preloadedState, store = configureStore({ reducer: { navSlice, formSlice, paletteSlice }, preloadedState }),
        ...renderOptions } = {}
) => {

    function Wrapper({ children }) {
        return (
            <BrowserRouter>
                <Provider store={store}>{children}</Provider>
            </BrowserRouter>
        );
    }

    return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
};

// re-export everything
export * from '@testing-library/react';
// override render method
export { render };