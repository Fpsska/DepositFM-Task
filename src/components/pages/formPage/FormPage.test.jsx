import { fireEvent } from '@testing-library/react';

import userEvent from '@testing-library/user-event';

import { render, screen } from '../../../test-utils';

import FormPage from './FormPage';


// /. imports


describe('FormPage component', () => {
    test('render component snapshot', () => {
        const { container } = render(<FormPage />);

        expect(container).toMatchSnapshot();
    });
    test('hide Preloader component, when isPreloaderVisible is false', () => {
        render(<FormPage />);

        expect(screen.queryByTestId('form-preloader')).toBeNull();  // not.toBeInTheDocument
        expect(screen.getByTestId('form-submit')).toHaveTextContent('Save');
    });
    test('show Preloader component, when isPreloaderVisible is true', async () => {
        render(<FormPage />);

        userEvent.click(screen.getByTestId('form-submit'));
        expect(screen.getByTestId('form-submit')).toHaveTextContent('Save');
        // expect(await screen.findByTestId('form-preloader')).toBeInTheDocument();
    });
});