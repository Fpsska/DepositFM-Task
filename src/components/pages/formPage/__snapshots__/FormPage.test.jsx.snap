// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormPage component render component snapshot 1`] = `
<div>
  <div
    class="form-page"
  >
    <div
      class="form-page__wrapper"
    >
      <div
        class="form-page__form"
      >
        <form
          class="form"
        >
          <div
            class="form__wrapper"
          >
            <div
              class="form__field field"
            >
              <div
                class="field__title"
              >
                <label
                  class="field__label"
                  for="name"
                >
                  Name
                </label>
                <span
                  class="field__warn"
                >
                  field cannot be empty
                </span>
              </div>
              <input
                class="field__input no-valid"
                id="name"
                name="name"
                placeholder="Your Name"
                required=""
                type="text"
                value=""
              />
            </div>
            <div
              class="form__field field"
            >
              <div
                class="field__title"
              >
                <label
                  class="field__label"
                  for="name"
                >
                  Surname
                </label>
                <span
                  class="field__warn"
                >
                  field cannot be empty
                </span>
              </div>
              <input
                class="field__input no-valid"
                id="surname"
                name="surname"
                placeholder="Your Surname"
                required=""
                type="text"
                value=""
              />
            </div>
            <div
              class="form__field field"
            >
              <div
                class="field__title"
              >
                <label
                  class="field__label"
                  for="name"
                >
                  Patronymic
                </label>
                <span
                  class="field__warn"
                >
                  field cannot be empty
                </span>
              </div>
              <input
                class="field__input no-valid"
                id="patronymic"
                name="patronymic"
                placeholder="Your Patronymic"
                required=""
                type="text"
                value=""
              />
            </div>
            <div
              class="form__field file"
            >
              <div
                class="file__title"
              >
                <span
                  class="file__text"
                >
                  Photo
                </span>
                <span
                  class="file__warn"
                >
                  image is required
                </span>
              </div>
              <input
                accept="image/*"
                class="file__input"
                id="file"
                name="file"
                required=""
                type="file"
              />
              <label
                class="file__label"
                for="file"
              />
            </div>
            <button
              class="button button--form"
              data-testid="button-submit"
              disabled=""
            >
              Save
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
